### PR TITLE
Rename Azure.Core.Extensions to Microsoft.Extensions.Azure

### PR DESCRIPTION
**Pull request by pakrym**
to align naming with other Microsoft.Extension integration libraries

### CODE REVIEW

File: sdk/core/Microsoft.Extensions.Azure/samples/Startup.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Azure;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
```
Comment (jsquire) [inline sdk/core/Microsoft.Extensions.Azure/samples/Startup.cs]: You may have intended to rename the namespace  in this file.

File: sdk/core/Microsoft.Extensions.Azure/src/Internal/AzureClientBuilder.cs [hunk 2]
```diff
@@ -1,9 +1,10 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using Azure.Core.Extensions;
 using Microsoft.Extensions.DependencyInjection;
 
-namespace Azure.Core.Extensions
+namespace Microsoft.Extensions.Azure.Internal
```
Comment (jsquire) [inline sdk/core/Microsoft.Extensions.Azure/src/Internal/AzureClientBuilder.cs]: Did you mean to add `Internal` to the namespaces in this directory?

File: sdk/core/Microsoft.Extensions.Azure/src/Microsoft.Extensions.Azure.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;$(RequiredTargetFrameworks)</TargetFrameworks>
```
Comment (jsquire) [inline sdk/core/Microsoft.Extensions.Azure/src/Microsoft.Extensions.Azure.csproj]: Is this needed now that you've marked the project as `IsClientLibrary`?    It looks like `Directory.Build.Data.props (line 53)` adds `netstandard2.0` as a common target.

**Review by jsquire — APPROVED** [review]
Did you intend to add `msbuild.binlog` to source control?

File: sdk/core/Microsoft.Extensions.Azure/src/Microsoft.Extensions.Azure.csproj (hunk 3 above)
Comment (pakrym) [reply sdk/core/Microsoft.Extensions.Azure/src/Microsoft.Extensions.Azure.csproj]: My bad, debugging leftovers.

**Review by tg-msft — APPROVED** [review]

**Pull request merged**